js/ui/codeView.js

```diff
@@ -0,0 +1,1623 @@
+    } catch (err) {
+        if (!err.matches(GLib.KeyFileError, GLib.KeyFileError.KEY_NOT_FOUND) &&
+            !err.matches(GLib.KeyFileError, GLib.KeyFileError.GROUP_NOT_FOUND))
+            logError(err, 'Error with block list in hack data file');
+        blockList = [];
+    }
+
+    return blockList.includes(app_id);
+}
+
+function _appIsAllowedHacking(desktopId) {
+    const keyfile = _ensureHackDataFile();
+    if (keyfile === null)
+        return true;
+
+    const appId = desktopId.slice(0, -8);  // remove ".desktop"
+    let allowOnlyList;
+    try {
+        [allowOnlyList] = keyfile.get_string_list('flip-to-hack', 'whitelist');
+    } catch (err) {
+        if (!err.matches(GLib.KeyFileError, GLib.KeyFileError.KEY_NOT_FOUND) &&
+            !err.matches(GLib.KeyFileError, GLib.KeyFileError.GROUP_NOT_FOUND))
+            logError(err, 'Error with allow-only list in hack data file');
+        allowOnlyList = [];
+    }
+
+    if (allowOnlyList.length === 0)
+        return true;
+    return allowOnlyList.includes(appId);
+}
+
+// _synchronizeMetaWindowActorGeometries
+//
+// Synchronize geometry of MetaWindowActor src to dst by
+// applying both the physical geometry and maximization state.
+function _synchronizeMetaWindowActorGeometries(src, dst) {
+    let srcGeometry = src.meta_window.get_frame_rect();
+    let dstGeometry = dst.meta_window.get_frame_rect();
+
+    let srcIsMaximized = (src.meta_window.maximized_horizontally &&
+                          src.meta_window.maximized_vertically);
+    let dstIsMaximized = (dst.meta_window.maximized_horizontally &&
+                          dst.meta_window.maximized_vertically);
+    let maximizationStateChanged = srcIsMaximized != dstIsMaximized;
+
+    // If we're going to change the maximization state, skip
+    // effects on the destination window, since we're synchronizing it
+    if (maximizationStateChanged)
+        Main.wm.skipNextEffect(dst);
+
+    if (!srcIsMaximized && dstIsMaximized)
+        dst.meta_window.unmaximize(Meta.MaximizeFlags.BOTH);
+
+    if (srcIsMaximized && !dstIsMaximized)
+        dst.meta_window.maximize(Meta.MaximizeFlags.BOTH);
+
+    if (!srcGeometry.equal(dstGeometry))
+        dst.meta_window.move_resize_frame(true,
+                                          srcGeometry.x,
+                                          srcGeometry.y,
+                                          srcGeometry.width,
+                                          srcGeometry.height);
+}
+
+function _synchronizeViewSourceButtonToRectCorner(button, rect) {
+    button.set_position(rect.x,
+                        rect.y + (rect.height - button.height) / 2);
+}
+
+function _getViewSourceButtonParams(interactive) {
+    return {
+        style_class: 'view-source',
+        x_fill: true,
+        y_fill: true,
+        reactive: interactive,
+        can_focus: interactive,
+        track_hover: interactive,
+        clip_to_allocation: true
```

Comment:
nitpick: To match the coding style use a comma at the end of this line (same comment in a couple of other locations; there should be an eslint config file in the root of the gnome-shell repo now, so you might want to run `eslint --fix` on the Hack-specific files)

Reply:
Done, I've ran the eslint and fixed all the issues in files that are added, the codeView.js, soundServer.js and the clubhouse.js.